### ISSUE TITLE
Keep keys quoted in JSON5?

### ISSUE THREAD

**Issue by suchipi** [prettier/prettier#4639]
Context: https://github.com/prettier/prettier/issues/4636#issuecomment-394515965 and https://github.com/prettier/prettier/issues/4636#issuecomment-394546383

Should we change the JSON5 printer to not convert object keys to identifiers?

I'm thinking yes, since JSON5 support in editors is hit or miss- for example, VS Code treats .babelrc as a JSON file instead of JSON5.

Personally, the only JSON5 feature I've ever leveraged was comments. But that's only one data point.

**Comment by j-f1** [comment]
IMO we should keep it as-is and fix the bugs in editors.

**Comment by aerze** [comment]
@suchipi I would like to put a vote in for keeping it as old JSON, or the least keeping `.babelrc` as JSON.

It really is surprising/frustrating when the editor _and_ the babel docs, [here for example](https://babeljs.io/docs/en/babel-preset-env#export-with-various-targets) still default to "normal" JSON.

**Comment by bakkot** [comment]
When people write JSON5 (in a way which makes it invalid JSON), do they tend to omit the quotes? Is there a way that could be assessed?

**Comment by jordanbtucker** [comment]
> When people write JSON5 (in a way which makes it invalid JSON), do they tend to omit the quotes? Is there a way that could be assessed?

Hi, I'm the current maintainer for JSON5. To answer your question, it's a bit of a mixed bag. The official JSON5 library outputs unquoted keys where possible, and single quoted otherwise.

When handwriting JSON5, I prefer to keep all of my keys unquoted if possible, but quote all of my keys if at least one of them must be quoted, just for consistency. I would say that most JSON5 documents are able to just use unquoted keys. (i.e. most keys in JSON and JSON5 documents are valid identifiers.)

My opinion would be to keep unquoted keys unquoted, and change the quote character for quoted keys according to the config file. For example, with the default config:

```json5
{ a: 1, 'b': 2, "c": 3 }
```

becomes

```json5
{ a: 1, "b": 2, "c": 3 }
```

**Comment by thorn0** [comment]
@jordanbtucker In other words, you propose that the default value for the `--quote-props` option should be change to `preserve` (currently it's `as-needed`, same as for JS). Right?

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwAEBDJaCMAaNAcgCNDsAmAgHXBuwGY0BfEPECABxgEtoBnZKAwAnYRADuABREIBKDABtxGAJ4C2xYRjABrODADKHbdygBzZDGEBXOGwAWMALYKA6ve7w+xsHAOzPbgA3TxVkcD51EFM+OGEYSS0zJwxkADNFWLYAKz4ADwAhLV19AwwnOAAZUzh0zLsQXLyDUzMFOABFawh4OoUskGNhWOFw3OgAVlZB4VMYV24AExh7ZAAOAAY2DjFY1y0OcJ24EaDatgBHbvhEzjkZk7iz6eE4K+5XxIxk1KQM-oasSc3EsNkBrXaXR6tT+9TYMAwxAWy1WSEoICsGG4ClaAGEIE4UuETlM2NZYgAVRFyf4DIK2ACSUEWCEMYFmXAAgsyDDAVO0+rEmEwgA)
```sh
--parser json5
--quote-props preserve
```

**Input:**
```json5
{ a: 1, 'b': 2, "c": 3 }
```

**Output:**
```json5
{ a: 1, "b": 2, "c": 3 }

```

**Comment by jordanbtucker** [comment]
@thorn0 I suppose so. Although I personally wouldn't want that to be the default for my JS files, haha.

**Comment by thorn0** [comment]
This issue was opened when the `--quote-props` option didn't exist. It was [added in April 2019](https://prettier.io/blog/2019/04/12/1.17.0.html#add-an-option-to-modify-when-prettier-quotes-object-properties-5934-by-azz) because it was needed for Closure Compiler. Now that we have it, this issue is probably not an issue anymore. People can use the option to adjust printing of JSON5 to their liking. Still, it might make sense to change this default (for JSON5 only, we aren't talking about JS here) and fix https://github.com/prettier/prettier/issues/5708#issuecomment-510571797 to make `--parser json5` "just work" out of the box for VS Code's "JSON with comments".

**Comment by thorn0** [comment]
Changing the default would be a breaking change though. I'm not sure what exactly it can break in this specific case, but usually we consider such changes breaking and that they need a major version bump. So it's probably not worth the trouble. Prettier has the option now, those who need it can use it.

**Comment by jordanbtucker** [comment]
Makes sense to me. Is there a way to mark this change to be added in the next major version?

**Comment by fisker** [comment]
I'm going to postpone this, since it's not been discussed for a long time.
